cmd/create.go

```diff
@@ -295,15 +295,27 @@ func checkOverwriteAllowed(depDir string, bp config.Blueprint, overwriteFlag boo
 		return nil // all good, no previous deployment
 	}
 
-	useWErr := fmt.Errorf("deployment folder %q already exists, use -w to overwrite", depDir)
+	if _, err := os.Stat(modulewriter.HiddenGhpcDir(depDir)); os.IsNotExist(err) {
+		// hidden ghpc dir does not exist
+		return fmt.Errorf("folder %q already exists, and it is not a valid GHPC deployment folder", depDir)
+	}
+
 	// try to get previous deployment
 	expPath := filepath.Join(modulewriter.ArtifactsDir(depDir), modulewriter.ExpandedBlueprintName)
+	if _, err := os.Stat(expPath); os.IsNotExist(err) {
+		return fmt.Errorf("expanded blueprint file %q is missing, maybe your previous GHPC version is too old", expPath)
```

Comment:
nit: I know you are using the verbiage from my comment but the combination of "maybe" and "too old" look less professional than I would like.

Could we instead say "expanded blueprint file %q is missing, this could be a result of changing GHPC version between consecutive deployments."